test/integration/connection_test.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from unittest import mock
+
+from pyexasol.exceptions import ExaConnectionDsnError
+from pyexasol.connection import ResolvedHost
+
+# pylint: disable=protected-access/W0212
+
+def test_resolve_hostname(connection):
+    with mock.patch("socket.gethostbyname_ex") as get_hostname:
+        get_hostname.return_value = ("host", [], ["ip1", "ip2"])
+        actual = connection._resolve_hostname("host", 1234, "fingerprint")
+        expected = [("host","ip1", 1234, "fingerprint"),("host","ip2", 1234, "fingerprint")]
+        assert actual == expected
+
+
+@pytest.mark.parametrize("empty_dsn", [None, "", " ", "\t"])
+def test_process_empty_dsn_fails(connection, empty_dsn):
+    with pytest.raises(ExaConnectionDsnError, match="Connection string is empty"):
+        connection._process_dsn(empty_dsn)
+
+def test_process_dsn_shuffles_hosts(connection):
+    dsn = "host1:1234,host2:4321"
+    def resolve_hostname():
+        with mock.patch("socket.gethostbyname_ex") as get_hostname:
+            get_hostname.side_effect = [("host1", [], ["ip11", "ip12"]), ("host2", [], ["ip21", "ip22"]),
+                                        ("host1", [], ["ip11", "ip12"]), ("host2", [], ["ip21", "ip22"])]
+            return tuple(connection._process_dsn(dsn))
+    count = 100
+    results = {resolve_hostname() for _ in range(0, count)}
+    assert len(results) > 1
+
+def test_process_dsn_without_shuffling(connection):
```

Comment:
consider passing in a patched connection (making use of the mock) so you can more cleanly separate the setup from the actual test.